nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -210,13 +210,21 @@ in
   options = {
     boot.zfs = {
       package = mkOption {
-        readOnly = true;
         type = types.package;
         default = if config.boot.zfs.enableUnstable then pkgs.zfsUnstable else pkgs.zfs;
         defaultText = literalExpression "if config.boot.zfs.enableUnstable then pkgs.zfsUnstable else pkgs.zfs";
         description = lib.mdDoc "Configured ZFS userland tools package.";
       };
 
+      modulePackage = mkOption {
+        type = types.package;
```

Comment:
This is necessary to enable users to choose their "kernel module" package.

Reply:
Maybe the best way to go about it is to implement a passthru that reimports the zfs derivation in kernel mode and just reuse the passthru to avoid any possible divergence.

Reply:
I don't quite understand why you want different kernel and userspace zfs versions?

Reply:
I am wondering if we should just drop the enableUnstable flag eventually and just have this option?

Reply:
It sounds good to me, but do we have any situation where the ZFS package and module package can be meaningfully different?

Reply:
I can make it internal and do the smart select inside the NixOS module.